ui/src/components/ContextMenu/index.tsx

```diff
@@ -1,10 +1,23 @@
 import type { Node } from "@flow/types";
 
-export type ContextMenuMeta = {
-  node?: Node;
+type ContextMenuStyles = {
   styles: React.CSSProperties;
 };
 
+type NodeContextMenuMeta = {
+  type: "node";
+  data: Node;
+};
+
+type SelectionContextMenuMeta = {
+  type: "selection";
+  data: Node[];
+};
```

Comment:
is `type: "node" | "nodes"` being used at all inside the component? It doesn't look like it, so if not you can just do something like this instead:
```
type ContextMenuMeta<T = any> = {
  data: T;
}
```
Then use the ContextMenu like 
```
<ContextMenu<Node[]> items={menuItems} contextMenuMeta={contextMenu} />;
```